src/response.rs

```diff
@@ -72,7 +76,58 @@ pub enum Response {
     Mutation(Result<()>),
 }
 
-use std::fmt;
+#[derive(Debug, PartialEq)]
+pub enum TryFromError {
+    WrongType,
+    Response(Error),
+}
+
+macro_rules! try_from {
+    ($ok_type:ty, $($variant:ident),*) => {
+        impl TryFrom<Response> for $ok_type {
+            type Error = TryFromError;
+            fn try_from(response: Response) -> std::result::Result<Self, Self::Error> {
+                match response {
+                    $(
+                        Response::$variant(Ok(data)) => Ok(data),
+                        Response::$variant(Err(error)) => Err(TryFromError::Response(error)),
+                    )*
+                    _ => Err(TryFromError::WrongType),
+                }
+            }
+        }
+    };
+}
+
+try_from!(IData, GetIData);
+try_from!(MData, GetMData, GetMDataShell);
+try_from!(u64, GetMDataVersion);
+try_from!(BTreeMap<Vec<u8>, Vec<u8>>, ListUnseqMDataEntries);
+try_from!(BTreeMap<Vec<u8>, MDataValue>, ListSeqMDataEntries);
+try_from!(BTreeSet<Vec<u8>>, ListMDataKeys);
+try_from!(Vec<MDataValue>, ListSeqMDataValues);
+try_from!(Vec<Vec<u8>>, ListUnseqMDataValues);
+try_from!(MDataPermissionSet, ListMDataUserPermissions);
+try_from!(BTreeMap<PublicKey, MDataPermissionSet>, ListMDataPermissions);
+try_from!(MDataValue, GetSeqMDataValue);
+try_from!(Vec<u8>, GetUnseqMDataValue, GetADataValue);
+try_from!(AData, GetAData, GetADataShell);
+try_from!(ADataOwner, GetADataOwners);
+try_from!(ADataEntries, GetADataRange);
+try_from!(ADataIndices, GetADataIndices);
+try_from!((Vec<u8>, Vec<u8>), GetADataLastEntry);
```

Comment:
Would be nice to define a type `Entry = (Vec<u8>, Vec<u8>)` in append_only_data.rs, exporting it as `ADataEntry`.

Reply:
If we're exporting that type anyway, do you mind if I just make it a proper struct, like
```rusr
struct Entry {
    pub key: Vec<u8>,
    pub value: Vec<u8>,
}
```

Reply:
Sure, that's even better than my suggestion 👍